src/build/plan.rs

```diff
@@ -352,6 +358,16 @@ impl JobQueue {
                 .map(|x| x.into_string().unwrap())
                 .collect();
 
+            // Send a window/progress notification. At this point we know the percentage
+            // started out of the entire cached build.
+            // FIXME. We could communicate the "program" being built here, but
+            // it seems window/progress notification should have message OR percentage.
+            {
+                let percentage = compiler_messages.len() as f64 / self.0.len() as f64;
```

Comment:
Since we're dividing by vector length here, it also might be a good idea to change the above `assert!(self.0.is_empty() == false);` to equivalent `assert!(self.0.len() > 0)` to better see that it's not 0 at a quick glance